doc/source/ray-contribute/getting-involved.rst

```diff
@@ -147,24 +147,76 @@ For Python documentation, we follow a subset of the `Google pydoc format <https:
         Do not introduce multi-line first sentences.
 
         Examples:
-            >>> # Provide code examples as possible.
-            >>> ray_canonical_doc_style(41, "hello")
-            True
+            .. doctest::
 
-            >>> # A second example.
-            >>> ray_canonical_doc_style(72, "goodbye")
-            False
+                >>> # Provide code examples for key use cases, as possible.
+                >>> ray_canonical_doc_style(41, "hello")
+                True
+
+                >>> # A second example.
+                >>> ray_canonical_doc_style(72, "goodbye")
+                False
 
         Args:
             param1: The first parameter. Do not include the types in the
-                docstring (they should be defined only in the signature).
+                docstring. They should be defined only in the signature.
                 Multi-line parameter docs should be indented by four spaces.
             param2: The second parameter.
 
         Returns:
             The return value. Do not include types here.
         """
 
+.. testcode::
+
+    class RayClass:
+        """The summary line for a class docstring should fit on one line.
+
+        Additional explanatory text can be added in paragraphs such as this one.
+        Do not introduce multi-line first sentences.
+
+        The __init__ method is documented here in the class level docstring.
+
+        All the public methods and attributes should have docstrings.
+
+        Examples:
+            .. testcode::
```

Comment:
is this nested `testcode` here on purpose?

Reply:
wow, if this works, it's cool!

Reply:
It's on purpose but I don't think the second testcode will be tested.

Reply:
@jjyao but then we should probably add another test that actually tests class creation, right?

Reply:
The second testcode basically just shows you how to write testcode in the docstring. I think it's fine that we don't test it? cc @bveeramani

Reply:
I think this is fine as a one-off thing.